### DISCUSSION TITLE
A Batch insert With no exceptions, But return error list , Is it Available ?

### DISCUSSION THREAD

**Discussion by muscodev** [psycopg/psycopg2#1648] [Q&A]
I would like to know that below scenario can be achieved by psycopg2

I have a dictionary of data to batch insert to postgresdb.
i will pass the prepared statement and this data to psycopg2 batch function, along with some argument saying ,'i want to commit all succesfully inserted, and rollback which got error'
finally return the error list if exist

similar functionality was found in the [oracle](https://cx-oracle.readthedocs.io/en/latest/user_guide/batch_statement.html)

**Comment by dvarrazzo** [comment]
Please look at psycopg 3 features about prepared statements and pipeline mode.

If the error you are talking about is a duplicate key you can use `INSERT ... OR CONFLICT DO NOTHING`.

Solving it in a generic case, whatever you do you need to set up a transaction per statement, so it will be inevitably slow, even if you use prepared statements and server stored procedures.
